### ISSUE TITLE
navigator.userAgentData not supported on Opera

### ISSUE THREAD

**Issue by sidewayss** [mdn/browser-compat-data#24526]
### What type of issue is this?

Incorrect support data (example: BrowserX says "86" but support was added in "40")

### What information was incorrect, unhelpful, or incomplete?

I have checked Opera on MacOS Sequoia and Windows 11. `navigator.userAgentData` is not there in the latest version 113 or version 111, which is the last version listed on [caniuse](https://caniuse.com/?search=useragentdata), though it lists support from v76-110 as well.  Here is a screenshot of Opera v111 on Win1. In the bottom left corner, in the Watch pane, outlined in red, is `navigator.userAgentData == undefined`.  In the upper right corner, outlined in red, is the Opera logo.

![userAgentData](https://github.com/user-attachments/assets/57544bb0-a83f-45cf-a65d-839fda8c8e99)

### What browsers does this problem apply to, if applicable?

Chromium (Chrome, Edge 79+, Opera, Samsung Internet)

### What did you expect to see?

That this feature is not supported in Opera.  I don't know if it used to be, but as of v111 it is not.

### Did you test this? If so, how?

browserstack

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgentData#browser_compatibility
https://caniuse.com/?search=useragentdata

### Do you have anything more you want to share?

In searching for existing, duplicate issues prior to submitting this one, I came across a failed attempt of mine relating to this same property.  I think this one is solid.  If there is agreement that this is incorrect information, with a pointer in the right direction I'm fine with making the change(s) and submitting the PR.

**Comment by queengooborg** [comment]
I just checked in BrowserStack and have confirmed that `navigator.userAgentData` is present and defined in Opera 113 on Windows 10 and macOS Sequoia:

<img width="489" alt="Screenshot 2024-09-25 at 12 08 07" src="https://github.com/user-attachments/assets/baa61ba8-f2cd-4eab-9044-40b2cbf3645b">

It looks like your issue is that you're not using an HTTPS page.  A secure context is required to use the feature.

**Comment by sidewayss** [comment]
I failed on this feature for a second time.  Sorry for the bother.  I thought "Naw, localhost can't have anything to do with supporting the feature."

**Comment by sidewayss** [comment]
...and on Chrome the feature is available for localhost.  I knew I was at least comparing apples to apples.  It's the browserstack localhost that is specifically insecure, same for Chrome on browserstack.  I'll have to install Opera to test it, and I'll have to push changes to a secure server every time I want to test on browserstack and have things look right.  I did not know that.

**Comment by queengooborg** [comment]
Normally in BrowserStack, on desktop platforms, you should be able to specify `localhost` instead of `bs-local.com`...!

**Comment by sidewayss** [comment]
This is the first time I've used BrowserStack on this relatively new Win11 install.  I think I used to be able to do that on Win10.  When I went to use the dashboard the other day, it forced me to install the "local" app, and the instructions said to use `bs-local.com`.  When I open a MacOS Sequoia/Safari session using `localhost` it says it can't find the server.  Maybe I should dig further and see if I can find a way to make it work with `localhost`...

**Comment by sidewayss** [comment]
fyi - I installed Opera locally and `userAgentData` is available and this returns true:
```js
navigator.userAgentData.brands.includes("Opera")
```

**Comment by sidewayss** [comment]
I fiddled with this a couple of times today and suddenly it's working, but I have no idea why.  The previous thing I did was a doomed-to-fail attempt using `https://localhost/etc.`.  Then I tried localhost again the normal way and voilà! browserstack connected for a change.

Chrome says the connection is insecure, but `navigator.userAgentData` is available, which is the same as connecting directly to my localhost without browserstack.

I searched online for more info and got little beyond the assertion that you can connect to localhost via https on browserstack.  Some of it was auto-testing features, not "Live".  But it was enough for me to fiddle with this stuff, and now I don't need bs-local.com anymore. I'm glad it's working in a more convenient way, but it doesn't exactly inspire confidence.

**Comment by sidewayss** [comment]
In case you care: `userAgentData` is acting erratically for me in BrowserStack on Android.  I'm testing on a Samsung S24 with Chrome.  Doesn't matter if it's localhost.

After failing to get it via an apparently secure localhost connection, I connected to github.com. Again `userAgentData` was undefined.  This had worked for me previously when localhost failed inexplicably, so I refreshed and otherwise fiddled around, to no avail.  After I arranged the widgets to take a screenshot with everything in view, including the device and browser name, `userAgentData` had a value in my Watch pane.  All of a sudden it appeared.

Is there a delay in acquiring the data as part of an async process?  I haven't bothered to read the detail about high/low entropy  and client hints, but it makes me wonder about the possibility of something async here.  The [MDN docs page](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgentData)s (and [here](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData)) do not contain the word "async".  I don't see anything anywhere about a callback/event upon completion.
